### PR TITLE
fix: fix lib/bot-deploy update not synchronize issue

### LINKED ISSUE #3311: when lib/bot-deploy update, it can not sync to plugins folder.
## Describe the bug

when lib/bot-deploy update, it can not sync to plugins folder(like azurePublish and azureFunctionsPublish) by running `yarn build` under composer folder.

FYI. 
import bot-deploy package in azurePublish by relative path.  `"@bfc/libs/bot-deploy": "../../packages/lib/bot-deploy"`. But it will copy bot-deploy package into node_modules under azurePublish folder in the first time, and not update it if yarn.lock exist.

## Browser

- [ ] Electron distribution
- [ ] Chrome
- [ ] Safari
- [ ] Firefox
- [ ] Edge

## OS

- [ ] macOS
- [ ] Windows
- [ ] Ubuntu

## Screenshots
image

### PR DESCRIPTION
**Pull request by VanyLaw**
## Description
Fix about using the relative path to install bot-deploy package under azurePublish and azureFunctionsPublish folder will not be update when lib/bot-deploy update.

Solution 1: 
remove yarn.lock under azurePublish and azureFunctionsPublish folder before yarn install can solve this problem. But It will cause yarn.lock change sometime. and maybe it will slow down yarn install.

Solution 2:
move bot-deploy package out of composer, because this package only be used by azurepublish plugin and azureFunctionpublish plugin. 

Solution 3:
Use relative path to import in code. like `import { BotProjectDeploy } from '../../../packages/lib/bot-deploy';`

Currently we take solution 1, but I am not sure which is better. @cwhitten @a-b-r-o-w-n  @benbrown Do you have any suggestion?

## Task Item

close #3311

### CODE REVIEW

**Comment by a-b-r-o-w-n** [comment]
The other option is to move the plugins inside of the yarn workspace which I think we wanted to do already. cc @benbrown @boydc2014

**Comment by a-b-r-o-w-n** [comment]
@VanyLaw Let's move all of the plugins into the yarn workspace. This will allow for a single lockfile and allow the plugins to have dependencies on other workspace packages.

So move `./Composer/plugins` into `./Composer/packages/plugins` and update everything that is necessary, including npm scripts. The naming convention should be `@bfc/plugin-<name>`, eg. (`@bfc/plugin-azure-publish`).

**Comment by VanyLaw** [comment]
Hi @a-b-r-o-w-n, I move plugins folder into packages, and update package name under `/plugins`. I e2e test localpublish and azurepublish with echoBot, it can work as before.

File: Composer/package.json [hunk 1]
```diff
@@ -38,11 +40,7 @@
     "build:server": "yarn workspace @bfc/server build",
     "build:client": "yarn workspace @bfc/client build",
     "build:tools": "yarn workspace @bfc/tools build:all",
-    "build:plugins": "yarn build:plugins:localpublish && yarn build:plugins:samples && yarn build:plugins:azurePublish && yarn build:plugins:azureFunctionsPublish",
-    "build:plugins:localpublish": "cd plugins/localPublish && yarn install && yarn build",
-    "build:plugins:samples": "cd plugins/samples && yarn install && yarn build",
-    "build:plugins:azurePublish": "cd plugins/azurePublish && yarn install && yarn build",
-    "build:plugins:azureFunctionsPublish": "cd plugins/azureFunctionsPublish && yarn install && yarn build",
+    "build:plugins": "yarn workspace @bfc/plugins build:all",
```
[the remark this answers is not in the document]
Comment (VanyLaw) [reply Composer/package.json]: done

**Comment by a-b-r-o-w-n** [comment]
There are still some `yarn.lock` files hanging around and a dotnet unit test is failing.

**Comment by VanyLaw** [comment]
> There are still some `yarn.lock` files hanging around and a dotnet unit test is failing.

Oh, I did haven't removed all the `yarn.lock` in plugin, because I think yarn.lock will be created automatically in every installation. But now I remove all the `yarn.lock` under plugins folder and add it into `.gitignore`

**Comment by a-b-r-o-w-n** [comment]
When using workspaces only the top-level `yarn.lock` file will be created and updated. No need to ignore the lock files since those would only every get created in an error.

**Comment by VanyLaw** [comment]
> When using workspaces only the top-level `yarn.lock` file will be created and updated. No need to ignore the lock files since those would only every get created in an error.

Thanks @a-b-r-o-w-n , I remove that in .gitignore under plugins folder now.

**Comment by zidaneymar** [comment]
dotnet build --runtime linux-x64 runtime\dotnet\
image

dotnet build is working correctly for both windows and linux runtime but LGTM check isn't passing.

**Comment by VanyLaw** [comment]
Hi @a-b-r-o-w-n, we run `dotnet build` and `dotnet test` in local, it can work well, but don't know why the LGTM analysis is failed. Could you please help to take a look and merge it?

**Comment by luhan2017** [comment]
I searched in the stackoverflow and it seems relate to the Ubuntu version. https://stackoverflow.com/questions/57790510/msbuild-unhandled-exception-the-filename-property-should-not-be-a-directory-unl

**Comment by a-b-r-o-w-n** [comment]
I'm inclined to take this since we don't control the build process in LGTM and the new code scanning action covers the same.

**Pull request merged**